### PR TITLE
[#6099] Remove low impact FxCop exclusions - Sample cases (1/2)

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses # 6099
#minor

## Description
This PR removes some of the FxCop rule exclusions with low impact.

### Detailed Changes
- **CA1819 - Properties should not return arrays**
   - Updated the `Records` property type in the `FeedbackRecords` class from array to collection.
- **CA1068 - CancellationToken parameters must come last**
   - Updated the `RecognizeAsync` method in the `Recognizer` class reordering the parameters.
- **CA1064 - Exceptions should be public**
   - Move protected class `InvokeResponseException` from the `ActivityHandler` class to its own file making it public.
- **CA1052 - Static holder types should be Static or NotInheritable**
   - Marked the `BotAssert` class as static.
- **CA1801 - Review unused parameters**
   - Removed unused CancellationToken parameter from `SetPropertyValueAsync` method of `BotState` class.
- **CA1034 - Nested types should not be visible**
   - Moved nested `CachedBotState` class to its own file.
- **CA1716 - Identifiers should not match keywords**
   - Renamed _next_ parameter of the `OnTurnAsync` method of `IMiddleware` interface as _nextDelegate_.
- **CA1055 - Uri return values should not be strings** 
   - Updated the `GetAttachmentUri` method of the `AttachmentEx` class to return Uri instead of string.
 - **CA1822 - Mark members as static**
   - Removing this rule exclusion in `TelemetryLoggerMiddleware` class didn't require changes.
- **CA1715 // Identifiers should have correct prefix**
   - Renamed generic type _TypeT_ as _TType_ in `GetChannelData` method in the `ActivityEx` class.
- **VSTHRD110 - Observe result of async calls** 
   - Added a variable to observe the result of Task.Run in `RequestManager` class.
- **VSTHRD200 - Use "Async" suffix for async methods**
   - Renamed the `Invoke` method of the `TelemetrySaveBodyASPMiddleware` class as `InvokeAsync`.

## Testing
This image shows the tests passing locally after the changes.
image

### CODE REVIEW

File: libraries/Microsoft.Bot.Schema/CardImage.cs [hunk 1]
```diff
@@ -30,9 +30,7 @@ public CardImage(string url = default, string alt = default, CardAction tap = de
         /// <summary>Gets or sets URL thumbnail image for major content property.</summary>
         /// <value>The URL of the thumbnail.</value>
         [JsonProperty(PropertyName = "url")]
-#pragma warning disable CA1056 // Uri properties should not be strings
-        public string Url { get; set; }
-#pragma warning restore CA1056 // Uri properties should not be strings
+        public Uri Url { get; set; }
```
Comment (JuanAr) [inline libraries/Microsoft.Bot.Schema/CardImage.cs:33]: Are there any Serialization tests for this class?
I'm concerned that the Bot Schema, which is used to render Activities must not be changed. At least, the serialized result should not change. 

https://docs.microsoft.com/en-us/azure/bot-service/rest-api/bot-framework-rest-connector-api-reference?view=azure-bot-service-4.0#cardimage-object
image

**Review by JuanAr — CHANGES_REQUESTED** [review]

**Review by JuanAr — CHANGES_REQUESTED** [review]
I'd only move the CardImage change to the PR with non recommended changes.

File: libraries/Microsoft.Bot.Connector/AttachmentsEx.cs [hunk 2]
```diff
@@ -32,21 +32,19 @@ public partial class Attachments
         /// <param name="attachmentId">id of the attachment.</param>
         /// <param name="viewId">default is "original".</param>
         /// <returns>uri.</returns>
-#pragma warning disable CA1055 // Uri return values should not be strings (we can't change this without breaking binary compat)
-        public string GetAttachmentUri(string attachmentId, string viewId = "original")
-#pragma warning restore CA1055 // Uri return values should not be strings
+        public Uri GetAttachmentUri(string attachmentId, string viewId = "original")
```
Comment (sw-joelmut) [inline libraries/Microsoft.Bot.Connector/AttachmentsEx.cs:35]: I was curious about how does he know that it has to return a Uri, and found this.
https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1055#cause

**Review by sw-joelmut — APPROVED** [review]

File: libraries/Microsoft.Bot.Schema/CardImage.cs (hunk 1 above)
Comment (ceciliaavila) [reply libraries/Microsoft.Bot.Schema/CardImage.cs]: Done. Thanks!

**Comment by ceciliaavila** [comment]
Promoted to MS in PR# 6162

**Pull request closed without merging**